### ISSUE TITLE
Objects that are created with `ValuData::from_json()` do not contain `prototype` instance (`__proto__`).

### ISSUE THREAD

**Issue by HalidOdat** [boa-dev/boa#452]
**Describe the bug**

Objects that are created with `ValuData::from_json()` do not contain `prototype` instance (`__proto__`).

**To Reproduce**
```js
JSON.parse("{x: 0}")
```

**Actual behavior**
```
{
	x: 0,
    extensible: true
}
```

**Expected behavior**
```
{
	x: 0,
    extensible: true,
	__proto__: {
		...  
	}
}
```

**Build environment (please complete the following information):**
 - OS: Manjaro Linux
 - Target triple: x86_64-unknown-linux-gnu
 - Rustc version:  rustc 1.43.1 (8d69840ab 2020-05-04)

**Additional context**

For this to be fixed we need `ValuData::from_json()` to take in an `Interpreter` (context) get the `prototype` and construct the `Object`.

**Comment by n14little** [comment]
@HalidOdat it looks like in order to get the `Interpreter` to `ValueData::from_json()` we would need to have access to the `Interpreter` in this conversion: https://github.com/boa-dev/boa/blob/bb2b6f638cba59269694d44424d16c9a5ede6a81/boa/src/builtins/value/conversions.rs#L191-L195
But, from my understanding, that is not possible. It seems that this conversion implementing `From<JSONValue> for Value` just have to go away? Is that correct, or is there something I'm missing? I'm pretty new to Rust.

**Comment by HalidOdat** [comment]
> But, from my understanding, that is not possible. It seems that this conversion implementing `From<JSONValue> for Value` just have to go away? Is that correct, or is there something I'm missing? I'm pretty new to Rust.

No. That's right. We should remove it and use `ValueData::from_json()` with it taking an `Interpreter`.

**Comment by n14little** [comment]
I am working on this to get myself going on some other issues assigned to me.
